devicetypes/sky-nie/min-smart-plug-dimmer.src/min-smart-plug-dimmer.groovy

```diff
@@ -0,0 +1,656 @@
+        if (cmd) {
+            result += zwaveEvent(cmd)
+        }
+        else {
+            logDebug "Unable to parse description: $description"
+        }
+
+        sendEvent(name: "lastCheckIn", value: convertToLocalTimeString(new Date()), displayed: false)
+    }
+    catch (e) {
+        log.error "$e"
+    }
+    return result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+    def encapCmd = cmd.encapsulatedCommand(commandClassVersions)
+
+    def result = []
+    if (encapCmd) {
+        result += zwaveEvent(encapCmd)
+    }
+    else {
+        log.warn "Unable to extract encapsulated cmd from $cmd"
+    }
+    return result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sensormultilevelv5.SensorMultilevelReport cmd) {
+    logTrace "SensorMultilevelReport: ${cmd}"
+
+    if (cmd.sensorValue != [255, 255]) { // Bug in beta device
+        switch (cmd.sensorType) {
+            case tempSensorType:
+                def unit = cmd.scale ? "F" : "C"
+                def temp = convertTemperatureIfNeeded(cmd.scaledSensorValue, unit, cmd.precision)
+
+                sendEvent(getEventMap("temperature", temp, true, null, getTemperatureScale()))
+                break
+            default:
+                logDebug "Unknown Sensor Type: ${cmd.sensorType}"
+        }
+    }
+    return []
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv1.ConfigurationReport cmd) {
+    logTrace "ConfigurationReport ${cmd}"
+
+    updateSyncingStatus()
+    runIn(4, refreshSyncStatus)
+
+    def param = configParams.find { it.num == cmd.parameterNumber }
+    if (param) {
+        def val = cmd.scaledConfigurationValue
+
+        logDebug "${param.name}(#${param.num}) = ${val}"
+        setParamStoredValue(param.num, val)
+    }
+    else {
+        logDebug "Parameter #${cmd.parameterNumber} = ${cmd.configurationValue}"
+    }
+    return []
+}
+private updateSyncingStatus() {
+    sendEventIfNew("syncStatus", "Syncing...", false)
+}
+
+def refreshSyncStatus() {
+    def changes = pendingChanges
+    sendEventIfNew("syncStatus", (changes ?  "${changes} Pending Changes" : "Synced"), false)
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+    logDebug "Ignored Command: $cmd"
+    return []
+}
+
+private getEventMap(name, value, displayed=null, desc=null, unit=null) {
+    def isStateChange = (device.currentValue(name) != value)
+    displayed = (displayed == null ? isStateChange : displayed)
+    def eventMap = [
+            name: name,
+            value: value,
+            displayed: displayed,
+            isStateChange: isStateChange,
+            descriptionText: desc ?: "${device.displayName} ${name} is ${value}"
+    ]
+
+    if (unit) {
+        eventMap.unit = unit
+        eventMap.descriptionText = "${eventMap.descriptionText}${unit}"
+    }
+    if (displayed) {
+        logDebug "${eventMap.descriptionText}"
+    }
+    return eventMap
+}
```

Comment:
Implementing such method and using it once seems like an overkill. I guess you can just remove it.

Reply:
I delete code as you list, I'm not sure others.

Reply:
Yes，you are right ！